app/src/main/java/com/android/unio/model/authentication/AuthViewModel.kt

```diff
@@ -65,20 +79,21 @@ constructor(private val firebaseAuth: FirebaseAuth, private val userRepository:
         if (user.isEmailVerified) {
           userRepository.getUserWithId(
               user.uid,
-              {
-                _authState.value =
-                    if (it.firstName.isNotEmpty()) {
-                      Screen.HOME
-                    } else {
-                      Screen.ACCOUNT_DETAILS
-                    }
-              },
-              {
-                Log.e("UnioApp", "Error fetching account details: $it")
-                _authState.value = Screen.WELCOME
+              onSuccess = { _authState.value = Screen.HOME },
+              onFailure = { error ->
+                if (error is FirebaseFirestoreException &&
+                    error.code == FirebaseFirestoreException.Code.NOT_FOUND) {
+                  _authState.value = Screen.ACCOUNT_DETAILS
+                } else {
+                  _authState.value = Screen.WELCOME
```

Comment:
Comment this as we've discussed the logic of checking type of error, but for future reference knowing that this is if the user only partially initialized his account will be useful

Reply:
Yes you are right, it's a pretty messy handling of navigation.